addon-test-support/@ember/test-helpers/-internal/helper-hooks.ts

```diff
@@ -0,0 +1,62 @@
+type Hook = (...args: any[]) => void | Promise<void>;
+type HookUnregister = {
+  unregister: () => void;
+};
+
+const registeredHooks = new Map<string, Set<Hook>>();
+
+/**
+ *
+ * @param {string} helperName The name of the test helper in which to run the hook.
+ * @param {string} order The order, either `start` or `end`, to run the hook.
+ * @returns {string} The compound key for the helper.
+ */
+function getHelperKey(helperName: string, order: string) {
+  return `${helperName}:${order}`;
+}
+
+/**
+ * Registers a hook function to be run during the invocation of a test helper.
+ *
+ * @public
```

Comment:
Lets make this `@private` while we iterate here.

Reply:
No problem.